projects/plugins/jetpack/_inc/client/security/firewall.jsx

```diff
@@ -0,0 +1,305 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { __, _x, sprintf } from '@wordpress/i18n';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import CompactFormToggle from 'components/form/form-toggle/compact';
+import { FormFieldset, FormLabel } from 'components/forms';
+import {
+	getPlanClass,
+	getJetpackProductUpsellByFeature,
+	FEATURE_SECURITY_SCANNING_JETPACK,
+} from 'lib/plans/constants';
+import FoldableCard from 'components/foldable-card';
+import JetpackBanner from 'components/jetpack-banner';
+import { withModuleSettingsFormHelpers } from 'components/module-settings/with-module-settings-form-helpers';
+import { ModuleToggle } from 'components/module-toggle';
+import SettingsCard from 'components/settings-card';
+import SettingsGroup from 'components/settings-group';
+import InfoPopover from '../components/info-popover';
+import Textarea from '../components/textarea';
+import { createInterpolateElement } from '@wordpress/element';
+import { getProductDescriptionUrl } from 'product-descriptions/utils';
+import { getSitePlan } from 'state/site';
+import {
+	getWafBootstrapPath,
+	getWafHasRulesAccess,
+	getWafUiEnabled,
+	isFetchingWafSettings,
+} from '../state/firewall/reducer';
+import QueryWafSettings from '../components/data/query-waf-bootstrap-path';
+
+export const Firewall = class extends Component {
+	/**
+	 * Get options for initial state.
+	 *
+	 * @returns {{jetpack_firewall_ip_list: *, jetpack_firewall_share_data: *}}
+	 */
+	state = {
+		jetpack_firewall_ip_list: this.props.getOptionValue( 'jetpack_firewall_ip_list' ),
+		jetpack_firewall_ip_allow_list: this.props.getOptionValue( 'jetpack_firewall_ip_allow_list' ),
```

Comment:
Would you mind changing the option names to `jetpack_waf_ip_allow_list` and `jetpack_waf_ip_block_list`? After talking with @ArSn while looking at the backend site he reminded that we already have some options that start with jetpack_waf, so having some that starts with jetpack_firewall would be a bit odd. Sorry about asking this after agreeing with the naming. 😅

Reply:
Updated!